src/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Mail;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Threading
+{
+    public static class EventWaitHandleAcl
+    {
+        /// <summary>Creates a new <see cref="EventWaitHandle" /> instance, ensuring it is created with the specified event security.</summary>
+        /// <param name="initialState"><see langword="true" /> to set the initial state to signaled if the named event is created as a result of this call; <see langword="false" /> to set it to non-signaled.</param>
+        /// <param name="mode">One of the enum values that determines whether the event resets automatically or manually.</param>
+        /// <param name="name">The name, if the event is a system-wide synchronization event; otherwise, <see langword="null" /> or an empty string.</param>
+        /// <param name="createdNew">When this method returns, it is set to <see langword="true" /> if a local event was created (that is, if name is <see langword="null" /> or an empty string) or if the specified named system event was created; <see langword="false" /> if the specified named system event already existed. This parameter is passed uninitialized.</param>
+        /// <param name="eventSecurity">The Windows access control security to apply.</param>
+        /// <returns>An object that represents the named event wait handle.</returns>
+        /// <exception cref="ArgumentException">The length of the name exceeds the maximum limit.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="eventSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The <paramref name="mode" /> enum value was out of legal range.</exception>
+        /// <exception cref="DirectoryNotFoundException">Could not find a part of the path specified in <paramref name="name" />.</exception>
+        /// <exception cref="WaitHandleCannotBeOpenedException">A system-wide synchronization event with the provided <paramref name="name" /> was not found.
+        /// -or-
+        /// An <see cref="EventWaitHandle" /> with system-wide name <paramref name="name" /> cannot be created. An <see cref="EventWaitHandle" /> of a different type might have the same name.</exception>
+        public static unsafe EventWaitHandle Create(bool initialState, EventResetMode mode, string name, out bool createdNew, EventWaitHandleSecurity eventSecurity)
+        {
+            if (name.Length > Interop.Kernel32.MAX_PATH)
```

Comment:
name should be allowed to be null

Reply:
Fixing.

Reply:
Tbh, I'd rather not check for the length at all here. Have we tested to see that it won't work without this? Like, is it `MAX_PATH` + prefix? (e.g. `GLOBAL\ReallyLongName`) I'd prefer to just let Windows report a failure and not inadvertently block some scenario.

Reply:
Sure, that's another option. I'll add a unit test and see what exception is thrown by Windows, if any.

Reply:
Update: Windows does not seem to be throwing an exception if we choose a name that goes beyond MAX_PATH.

Reply:
@JeremyKuhne  We do throw in Framework. Should we match the behavior in .NET Core?
https://referencesource.microsoft.com/#mscorlib/system/threading/eventwaithandle.cs,59

Reply:
I'm probably going to answer my own question and say no, because we don't throw on MAX_LENGTH in the .NET Core constructor for EventWaitHandle:
https://source.dot.net/#System.Private.CoreLib/shared/System/Threading/EventWaitHandle.Windows.cs,20